### PR TITLE
Generate compatibility spinner bars from compatibility json file

### PR DESCRIPTION
**Pull request by Razzile**

### CODE REVIEW

File: docs/css/spinner.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    border-radius: 50%;
+    width: 120px;
+    height: 120px;
+    animation: spin 1.2s linear infinite;
+  }
+  
+  @keyframes spin {
+    0% { transform: rotate(0deg); }
+    100% { transform: rotate(360deg); }
+  }
```
Comment (Margen67) [inline docs/css/spinner.css:13]: eof newline

File: docs/javascripts/compatibility_list.js [hunk 2]
```diff
@@ -22,7 +22,98 @@
...
+    }
+
+    generateCompatibilityProgressBars(issues, compatMap);
+
+}).catch(e => {
+    // hide spinner and show error message
+    const loadingDiv = document.getElementsByClassName("progress-bars-loading")[0];
+    loadingDiv.firstElementChild.style.display = "none";
+    loadingDiv.lastElementChild.style.display = "block";
+});
```
Comment (Margen67) [inline docs/javascripts/compatibility_list.js:119]: eof newline

**Comment by halotroop2288** [comment]
While I trust your decision to use HTML instead of Markdown is for good reason, I'd really like to find a way to avoid it. [`md-in-html`] should allow you to tag MD with div classes so you can change the values inside it.

With all that said, I will merge this as soon as I have tested it. I'll do some more testing myself to see if it can be done my preferred way.

[`md-in-html`]:https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/?h=md+in+html#markdown-in-html

File: docs/javascripts/compatibility_list.js [hunk 3]
```diff
@@ -22,7 +22,98 @@
  SOFTWARE.
```
Comment (halotroop2288) [inline docs/javascripts/compatibility_list.js:22]: This script is mostly yours now. You may relicense and take credit for it in the header as you choose, @Razzile.

**Review by halotroop2288 — APPROVED** [review]

**Comment by halotroop2288** [comment]
@all-contributors add @Razzile for code

**Pull request merged**